pkg/expr/sql/frame_db_conv.go

```diff
@@ -159,6 +165,34 @@ func fieldValFromRowVal(fieldType data.FieldType, val interface{}) (interface{},
 	case data.FieldTypeBool, data.FieldTypeNullableBool:
 		return parseBoolFromInt8(val, nullable)
 
+	case data.FieldTypeJSON, data.FieldTypeNullableJSON:
+		switch v := val.(type) {
+		case types.JSONDocument:
+			// Use fmt.Sprintf to serialize JSONDocument to []byte
+			raw := json.RawMessage(v.String())
+			if nullable {
+				return &raw, nil
+			}
+			return raw, nil
+	
+		case string:
+			raw := json.RawMessage(v)
+			if nullable {
+				return &raw, nil
+			}
+			return raw, nil
+	
+		case []byte:
+			raw := json.RawMessage(v)
+			if nullable {
+				return &raw, nil
+			}
+			return raw, nil
+	
+		default:
+			return nil, fmt.Errorf("unexpected type for JSON field: %T", val)
```

Comment:
Surprised to see `%T` here for `val`, when we use `%v` for val four lines below this. I wonder if we can use the same in both places? I don't mind which we choose.